### ISSUE TITLE
There is no option available for SSL pining. Please provide a way for pining certificate.

### ISSUE THREAD

**Issue by VeeruDjaat** [SDWebImage/SDWebImage#2956]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 12.0 / 10.14.0 / 12.0 / 5.0
 SDWebImage Version      | e.g. 5.0.0 / 4.4.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 11 / Xcode 10
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by dreampiggy** [comment]
You can.

Use [URLSession:task:didReceiveChallenge:completionHandler:](https://developer.apple.com/documentation/foundation/nsurlsessiontaskdelegate/1411595-urlsession) the delegate from URLSession.

You need to subclass `SDWebImageDownloaderOperation` class, then override this method, and set the `operationClass` property of `SDWebImageDownloaderConfig`.

See wiki: https://github.com/SDWebImage/SDWebImage/wiki/Advanced-Usage#custom-download-operation-40

**Comment by dreampiggy** [comment]
You can also use `SDWebImageDownloaderAllowInvalidSSLCertificates` to bypass any SSL error.

**Comment by VeeruDjaat** [comment]
Thanks buddy for quick response. I went through wiki. I am confused from where (or how) to set the operationClass property of SDWebImageDownloaderConfig.  
I am using like - imageView.sd_setImage(with: URL(string: url), placeholderImage: UIImage(named: "imagePlaceholder"), options: .continueInBackground)

**Comment by dreampiggy** [comment]
Directly set it before any `sd` call. For example, the `applicationDidFinishLaunching:` time. :)

**Comment by VeeruDjaat** [comment]
Perfect ! Thanks a ton.
